crates/engine/tree/src/tree/payload_processor/mod.rs

```diff
@@ -68,9 +68,10 @@ where
     precompile_cache_disabled: bool,
     /// Precompile cache map.
     precompile_cache_map: PrecompileCacheMap<SpecFor<Evm>>,
-    /// A cleared sparse trie, kept around to be reused for the state root computation so that
-    /// allocations can be minimized.
-    sparse_trie: Option<SparseTrie>,
+    /// A cleared accounts sparse trie, kept around to be reused for the state root computation so
+    /// that allocations can be minimized.
+    accounts_trie_tx: mpsc::SyncSender<SparseTrie>,
+    accounts_trie_rx: mpsc::Receiver<SparseTrie>,
```

Comment:
from what I can tell, this is using a channel like a mutex?

this should not introduce any race condition issues because the trie must be empty here so it's either available or not

but should this be an `Arc<Mutex` instead?